Week-01/Day-01_Convert-Ages-To-Days/convert_ages_to_days_botiboti/src/main.rs

```diff
@@ -0,0 +1,61 @@
+use num_bigint::{BigUint, ParseBigIntError};
+use std::io;
+
+const DAYS_IN_YEAR: u64 = 365;
+
+// convert input string to BigUint
+fn read_parse(inp: &str) -> Result<BigUint, ParseBigIntError> {
+    let age = match inp.trim().parse::<BigUint>() {
+        Ok(age) => age,
+        Err(e) => return Err(e),
+    };
+    Ok(age)
+}
+
+// turn age in years to days
+fn calc_days(age: &BigUint) -> BigUint {
+    age * DAYS_IN_YEAR
+}
+
+fn main() {
+    println!("Enter age:");
+
+    let mut input = String::new();
+
+    io::stdin()
+        .read_line(&mut input)
+        .expect("Failed to read line.");
+
+    let age: BigUint = read_parse(&input).expect("Not a positive integer.");
+    println!("You entered: {}.", age);
+    println!("That's roughly {} days.", calc_days(&age));
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn an_ordinary_human_age() {
+        let result = read_parse(&String::from("50"));
+        assert_eq!(result, Ok(BigUint::from(50_u32)));
+    }
+
+    #[test]
+    fn age_of_earth() {
+        let result = read_parse(&String::from("4500000000"));
+        assert_eq!(result, Ok(BigUint::from(4500000000_u64)));
+    }
+
+    #[test]
+    #[should_panic]
+    fn not_a_positive_integer() {
+        read_parse(&String::from("boo")).unwrap();
+    }
+
+    #[test]
+    fn calc_days_for_earth() {
```

Comment:
we need more than one test for the actual function

Reply:
I will add for ages: 0, 1 and 100

Reply:
Yes, you can assert several things in one unit test "on the same action". But it does not apply here, since the "action" (i'd rather call it "result" or "value") is calling `calc_days` with a specific number, and calling it with another number is a different action.
Here's an example where we assert multiple properties on a prepared test value.
```
let result = calc_days(BigUint::from(365_u64)).unwrap());
assert!(result > 300);
assert(result < 500);
assert_eq!(result, BigUint::from(365_u64));
```
Triple A link is not really relevant, it suggest to separate the stages of a test, not to assert several properties on one. It actually warns of "Test methods that try to test too many different things at once".
In conclusion, I don't think we should put calls to `calc_days` with different arguments into a sigle unit test, this is fine as is.
Sctually this discussion revealed a problem with the tests, which is testing too many things at once, see below.

Reply:
@brainrake 
I get each of your points and agree that the Triple A link is not really proving my point. 

I however do not see why it would be a different action if the value of the function argument changes. To me it is just repetition. Also this way when looking at `cargo test` I do not know what action am I testing when I see `test tests::calc_zero` 

I want progress here so I am not going to block anyone, but I wanted to get my opinion through.

Reply:
An action is something you do, like a calculation. The result can have different properties (like being less than or more than a given number, etc.). Calling a function with a different argument will produce a different result, which will have different properties.
Sometimes it makes sense to have several calls, such as in property-based testing (like QuickCheck), where the function is called with many random values and the result is checked for correctness properties. Another example is a macro like `#[test_value 1 => 365]`(I saw this somewhere, don't rememeber the name) that makes a test that calls the function, and several example inputs and outputs can be given. This is not one of those times.
@Smorci please resolve this conversation and approve the PR if you're more convinced than not.